### PR TITLE
Use new actors API; Akka gRPC 1.0.0

### PR DESCRIPTION
**Pull request by ennru**
https://github.com/akka/akka-grpc/issues/1008

### CODE REVIEW

File: src/main/g8/src/main/java/com/example/helloworld/GreeterServer.java [hunk 1]
```diff
@@ -37,31 +38,31 @@ public static void main(String[] args) throws Exception {
     // important to enable HTTP/2 in ActorSystem's config
     Config conf = ConfigFactory.parseString("akka.http.server.preview.enable-http2 = on")
       .withFallback(ConfigFactory.load());
-    ActorSystem system = ActorSystem.create("HelloWorld", conf);
+    ActorSystem<Void> system = ActorSystem.create(Behaviors.empty(), "GreeterServer", conf);
```
Comment (patriknw) [inline src/main/g8/src/main/java/com/example/helloworld/GreeterServer.java:41]: I would have defined a Guardian Behavior and initialize http/grpc from there.

File: src/main/g8/src/test/java/com/example/helloworld/GreeterServiceImplTest.java [hunk 2]
```diff
@@ -15,22 +13,20 @@
 
 public class GreeterServiceImplTest {
 
-  private static ActorSystem system;
-  private static Materializer materializer;
+  private static final ActorTestKit testKit = ActorTestKit.create();
```
Comment (patriknw) [inline src/main/g8/src/test/java/com/example/helloworld/GreeterServiceImplTest.java:16]: wouldn't it be better to use ClassRule with TestKitJunitResource?

File: src/main/g8/src/test/java/com/example/helloworld/GreeterTest.java [hunk 3]
```diff
@@ -20,35 +21,34 @@
 
 public class GreeterTest {
 
-  private static ActorSystem serverSystem;
-  private static ActorSystem clientSystem;
+  // important to enable HTTP/2 in server ActorSystem's config
+  private static final Config config = ConfigFactory
+          .parseString("akka.http.server.preview.enable-http2 = on")
+          .withFallback(ConfigFactory.defaultApplication());
+  private static final ActorTestKit testKit = ActorTestKit.create(config);
```
Comment (patriknw) [inline src/main/g8/src/test/java/com/example/helloworld/GreeterTest.java:28]: same here

File: src/main/g8/src/test/java/com/example/helloworld/GreeterTest.java [hunk 4]
```diff
@@ -20,35 +21,34 @@
...
       );
-
   }
 
   @AfterClass
   public static void teardown() {
-    TestKit.shutdownActorSystem(serverSystem);
-    TestKit.shutdownActorSystem(clientSystem);
+    TestKit.shutdownActorSystem(Adapter.toClassic(clientSystem));
+    testKit.shutdownTestKit();
```
Comment (ennru) [inline src/main/g8/src/test/java/com/example/helloworld/GreeterTest.java:51]: What's our recommendation when we need two actor systems in a test? This shutdown looks very inconsistent...

Comment (patriknw) [reply src/main/g8/src/test/java/com/example/helloworld/GreeterTest.java:51]: I think it’s possible to have morethan one TestKitJunitResource

Comment (ennru) [reply src/main/g8/src/test/java/com/example/helloworld/GreeterTest.java]: Found `ActorTestkit.shutdown`.

File: .travis.yml [hunk 5]
```diff
@@ -1,30 +1,52 @@
...
+jobs:
+  include:
+    - stage: check
+      name: "sbt & Paradox (JDK 8)"
+      script: sbt test docs/paradox
+      env: TRAVIS_JDK=adopt@1.8.0-242
+
+    - name: "sbt & Paradox (JDK 11)"
+      script: sbt test docs/paradox
+      env: TRAVIS_JDK=adopt@1.11.0-6
```
Comment (ignasi35) [inline .travis.yml:18]:
```suggestion
      env: TRAVIS_JDK=adopt@1.11.0-7
```
(same in jobs below)

File: .travis.yml [hunk 6]
```diff
@@ -1,30 +1,52 @@
...
   - find $HOME/.ivy2 -name "ivydata-*.properties" -print -delete
   - find $HOME/.sbt  -name "*.lock"               -print -delete
 
 cache:
   directories:
+    - $HOME/.cache/coursier
     - $HOME/.ivy2/cache
     - $HOME/.sbt/boot
     - $HOME/.gradle
     - $HOME/.jabba/jdk
```
Comment (ignasi35) [inline .travis.yml:51]:
```suggestion
    - $HOME/.jabba
```

File: src/main/g8/pom.xml [hunk 7]
```diff
@@ -22,13 +28,20 @@
...
+      <url>https://dl.bintray.com/akka/snapshots</url>
+    </pluginRepository>
   </pluginRepositories>
 
   <properties>
     <maven.compiler.source>1.8</maven.compiler.source>
     <maven.compiler.target>1.8</maven.compiler.target>
     <akka.version>$akka_version$</akka.version>
-    <akka.grpc.version>$akka_grpc_version$</akka.grpc.version>
+    <akka.grpc.version>1.0.0-M1+35-c8aa943b</akka.grpc.version>
```
Comment (ignasi35) [inline src/main/g8/pom.xml:43]:
```suggestion
    <akka.grpc.version>1.0.0-M1+37-732f2dee</akka.grpc.version>
```

File: src/main/g8/project/plugins.sbt [hunk 8]
```diff
@@ -1,5 +1,7 @@
-addSbtPlugin("com.lightbend.akka.grpc" % "sbt-akka-grpc" % "$akka_grpc_version$")
-
+// addSbtPlugin("com.lightbend.akka.grpc" % "sbt-akka-grpc" % "$akka_grpc_version$")
 addSbtPlugin("com.lightbend.sbt" % "sbt-javaagent" % "0.1.4")
-
 addSbtPlugin("io.spray" % "sbt-revolver" % "0.9.1")
+
+// TODO remove when Akka gRPC 1.0.0 is final
+addSbtPlugin("com.lightbend.akka.grpc" % "sbt-akka-grpc" % "1.0.0-M1+35-c8aa943b")
```
Comment (ignasi35) [inline src/main/g8/project/plugins.sbt:6]:
```suggestion
addSbtPlugin("com.lightbend.akka.grpc" % "sbt-akka-grpc" % "1.0.0-M1+37-732f2dee")
```

**Comment by ignasi35** [comment]
I tried to sync the builds with https://github.com/akka/akka-grpc-quickstart-scala.g8/pull/43 I now get an error using Gradle. I've tried reproducing it in the scala version of the seed but it all works in that project.

**Comment by ignasi35** [comment]
I'm parking this for a while

**Comment by ignasi35** [comment]
@ennru as discussed, this is ready to be merged.

cc https://github.com/akka/akka-grpc/issues/1034

**Pull request merged**